src/wallet/scriptpubkeyman.cpp

```diff
@@ -1947,6 +1947,69 @@ std::optional<MigrationData> LegacyScriptPubKeyMan::MigrateToDescriptor()
     return out;
 }
 
+bool LegacyScriptPubKeyMan::DeleteRecords(bilingual_str& error)
+{
+    LOCK(cs_KeyStore);
+    WalletBatch batch(m_storage.GetDatabase());
```

Comment:
let's begin a db txn here and commit it at the end of the function.

otherwise if an error arises in the middle, we will end up with half of the information removed.

Reply:
Good point.